habitat-lab/habitat/sims/habitat_simulator/metadata_interface.py

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Meta Platforms, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import os
+from collections import defaultdict
+from typing import Dict, List, Optional, Union
+
+# TODO: this is not a lab dependency currently, should probably remove it
+import pandas as pd
+
+import habitat.datasets.rearrange.samplers.receptacle as hab_receptacle
+import habitat.sims.habitat_simulator.sim_utilities as sutils
+import habitat_sim
+from habitat_sim.physics import ManagedArticulatedObject, ManagedRigidObject
+
+
+class MetadataInterface:
```

Comment:
Should this be in Habitat-Lab codebase? This feels very application specific to me.

Reply:
Agreed. At this point it is only used map semantic classes to states. That doesn't need to be a base functionality. We could merge the machinery and use Habitat semantic_scene for tests. Then we'd extend the state classes to include what we want in downstream code.

Reply:
I've stripped this out for this current PR and unit tests.